plugins/polio/js/src/domains/LQAS-IM/LQAS/CountryOverview/utils.ts

```diff
@@ -0,0 +1,198 @@
+    roundNumber,
+}: {
+    data?: Record<string, LqasImCampaign>;
+    campaignObrName?: string;
+    roundNumber?: number;
+}): string => {
+    const checked = sumChildrenCheckedLqas(roundNumber, data, campaignObrName);
+    const notMarked = childrenNotMarked({
+        data,
+        campaign: campaignObrName,
+        round: roundNumber,
+    });
+    return convertStatToPercent(notMarked, checked);
+};
+
+/**
+ * Nfm stands for "No finger mark"
+ * This hook returns the title to be used in a chart
+ * showing data about children that were not vaccinated, i.e had no mark on their finger
+ */
+export const useLqasNfmTitle = ({
+    data,
+    campaignObrName,
+    roundNumber,
+}: {
+    data?: Record<string, LqasImCampaign>;
+    campaignObrName?: string;
+    roundNumber?: number;
+}): string => {
+    const { formatMessage } = useSafeIntl();
+    const ratioUnmarked = makeLqasRatioUnmarked({
+        data,
+        campaignObrName,
+        roundNumber,
+    });
+    return `${formatMessage(MESSAGES.childrenNoMark)}: ${ratioUnmarked}`;
+};
+
+type UseVerticalChartDataArgs = {
+    data?: Record<string, LqasImCampaign>;
+    campaignObrName?: string;
+    roundNumber?: number;
+    formatter: (args: FormatForNFMArgs<string>) => BarChartData[];
+    type: string;
+};
+
+export const useLqasVerticalChartData = ({
+    data,
+    campaignObrName,
+    formatter,
+    type,
+    roundNumber,
+}: UseVerticalChartDataArgs): BarChartData[] => {
+    const { formatMessage } = useSafeIntl();
+    return useMemo(() => {
+        return formatter({
+            data,
+            campaign: campaignObrName,
+            round: roundNumber,
+            formatMessage,
+            type,
+        });
+    }, [formatter, data, campaignObrName, roundNumber, formatMessage, type]);
+};
+
+export const useLqasRfaTitle = ({
+    data,
+    campaignObrName,
+    roundNumber,
+}: {
+    data?: Record<string, LqasImCampaign>;
+    campaignObrName?: string;
+    roundNumber?: number;
+}): string => {
+    const { formatMessage } = useSafeIntl();
+    const accessor = 'childabsent';
+    const childrenAbsent = calculateChildrenAbsent({
+        data,
+        campaign: campaignObrName,
+        round: roundNumber,
+        accessor,
+    });
+    return `${formatMessage(MESSAGES.childrenNfmAbsent)}: ${childrenAbsent}`;
+};
+
+export const findScopeIdsForRound = ({
+    campaign,
+    roundNumber,
+}: {
+    campaign: Campaign;
+    roundNumber?: number;
+}) => {
+    if (!campaign.separate_scopes_per_round) {
+        return campaign.scopes
+            ?.filter(scope => scope.group)
+            .map(scope => scope.group!.org_units)
+            .flat();
+    }
+    const fullRound = campaign.rounds?.find(
+        round => round.number === roundNumber,
+    );
+    if (fullRound) {
+        return fullRound.scopes
+            ?.filter(scope => scope.group)
+            .map(scope => scope.group!.org_units)
+            .flat();
+    }
+    return [];
```

Comment:
flatMap and useMemo ?

Reply:
TIL flatMap. No useMemo, because we're not in a hook, and the method is called in a useMemo hook anyway. seems more flexible that way